syntax = "proto3";

package protos;

service MonitorService {
    rpc Heartbeat (HeartbeatRequest) returns (HeartbeatOkResponse) {}
    rpc InstanceMetrics (InstanceMetricsRequest) returns (InstanceMetricsOkResponse) {}
}

message HeartbeatRequest {}

message HeartbeatOkResponse {
    bool status = 1;
}

message InstanceMetricsRequest {
    float prev = 1;
    string direction = 2;
}

message InstanceMetricsOkResponse {
    float cpu_load = 1;
}
